### PR TITLE
[Merged by Bors] - feat: use the function to help elaborate the injectivity lemma in `apply_fun`

### PR DESCRIPTION
**Pull request by alexjbest**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by alexjbest** [comment]
Not sure if this will conflict with #6732 but it shouldn't really matter which one gets applied first

File: test/apply_fun.lean [hunk 1]
```diff
@@ -225,11 +225,15 @@ example (f : α ≃ β) (x y : α) (h : f x = f y) : (fun s => s) (x = y) := by
...
+  apply_fun (Nat.cast : ℕ → ℚ) using Nat.cast_injective
+  rfl
+
 -- Check that locals are elaborated properly in apply_fun
 example : 1 = 1 := by
   let f := fun (x : Nat) => x + 1
   -- clearly false but for demo purposes only
   have g : ∀ f, Function.Injective f
   · sorry
   apply_fun f using (g f)
```
Comment (kmill) [inline test/apply_fun.lean:239]: Oops, sorry, I messed up merging.
```suggestion
  apply_fun f using (g f)
  rfl
```

**Comment by kmill** [comment]
It'd be better to throw an error when the `using` clause's lemma isn't actually used, but this PR is certainly an improvement. Please merge once CI succeeds.

bors d+

**Comment by alexjbest** [comment]
Thanks for merging @kmill, I definitely made too many apply_fun tweaks recently ;)

**Pull request closed without merging**